src/sourceParser.js

```diff
@@ -7,13 +7,53 @@
 
 const importParser = /\bimport\s*(?:\(|\/[/*])/;
 
-export function rejectImportExpressions(s) {
-  const index = s.search(importParser);
-  if (index !== -1) {
-    // todo: if we have a full parser available, use it here. If there is no
-    // 'import' token in the string, we're safe.
-    // if (!parse(s).includes('import')) return;
-    const linenum = s.slice(0, index).split('\n').length; // more or less
-    throw new SyntaxError(`possible import expression rejected around line ${linenum}`);
+export function parseSource(s, options) {
+  const { infixBangResolver } = Object(options);
+
+  // eslint-disable-next-line global-require
+  const parser = require('@agoric/babel-parser');
+  if (infixBangResolver) {
+    return parser.parse(s, {
+      plugins: [['infixBang', { resolver: infixBangResolver }]]
+    });
+  }
+  return parser.parse(s);
+}
+
+export function generateSource(s, ast) {
+  // eslint-disable-next-line global-require
+  const { default: generate } = require('@babel/generator');
```

Comment:
Oh, that's weird: mixing `require` from CJS modules with `export` from ESM modules in the same file. I think I see why: so you don't bring it in if it is not needed. Worth an inline comment.

What should we do long term? I would like to avoid using both kinds of modules, especially in the same file.